packages/enzyme/src/ShallowWrapper.js

```diff
@@ -399,11 +399,11 @@ class ShallowWrapper {
    * @param {Function} cb - callback function
    * @returns {ShallowWrapper}
    */
-  setProps(props, callback = noop) {
+  setProps(props, callback) {
     if (this[ROOT] !== this) {
       throw new Error('ShallowWrapper::setProps() can only be called on the root');
     }
-    if (typeof callback !== 'function') {
+    if (callback && typeof callback !== 'function') {
```

Comment:
if you pass `undefined`, the current code will replace it with `noop` - this would allow you to silently pass a falsy primitive (false, NaN, 0, empty string, null). What's the utility of this?

Reply:
What about `null`? The default won't kick in. React check both:
https://github.com/facebook/react/blob/5031ebf6beddf88cac15f4d2c9e91f8dbb91d59d/packages/react-reconciler/src/ReactFiberClassComponent.js#L179

```js
function b(b = 'b') {
  return b
}

b() // 'b'
b(null) // null
b(undefined) // 'b'
b(false) // false
```

Reply:
You're right that `null` won't trigger the default; but i'm confused why you'd ever pass in a null in the first place.

Reply:
> i'm confused why you'd ever pass in a null in the first place.

Good question, I have no idea. I was curious to see how React would believe, turns out, they explicitly support it for some reason I'm not aware of.

Reply:
I've asked for the motivation [here](https://github.com/facebook/react/commit/d3aeca1ff89e2ecedcd9be5ebe0a08336def5048#commitcomment-30185172)